src/main/kotlin/com/sourcegraph/cody/statusbar/CodyStatusBarActionGroup.kt

```diff
@@ -1,36 +1,50 @@
 package com.sourcegraph.cody.statusbar
 
 import com.intellij.ide.actions.AboutAction
-import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.sourcegraph.cody.config.CodyApplicationSettings
 import com.sourcegraph.config.ConfigUtil
 
 class CodyStatusBarActionGroup : DefaultActionGroup() {
   override fun update(e: AnActionEvent) {
     super.update(e)
     e.presentation.isVisible = ConfigUtil.isCodyEnabled()
-  }
 
-  override fun getChildren(e: AnActionEvent?): Array<AnAction> {
+    removeAll()
     if (CodyAutocompleteStatusService.getCurrentStatus() ==
-        CodyAutocompleteStatus.CodyAgentNotRunning)
-        return listOf(
-                OpenLogAction(),
-                AboutAction().apply {
-                  templatePresentation.text = "Open About To Troubleshoot Issue"
-                },
-                ReportCodyBugAction())
-            .toTypedArray()
+        CodyAutocompleteStatus.CodyAgentNotRunning) {
+      addAll(
+          listOf(
+              OpenLogAction(),
+              AboutAction().apply {
+                templatePresentation.text = "Open About To Troubleshoot Issue"
+              },
+              ReportCodyBugAction()))
+    } else {
+
+      val warningActions =
+          if (CodyApplicationSettings.instance.autocompleteRateLimitError &&
+              CodyApplicationSettings.instance.chatRateLimitError) {
+            ChatAndAutocompleteLimitWarningAction()
+          } else if (CodyApplicationSettings.instance.autocompleteRateLimitError) {
+            AutocompletionLimitWarningAction()
+          } else if (CodyApplicationSettings.instance.chatRateLimitError) {
+            ChatLimitWarningAction()
+          } else {
+            null
+          }
```

Comment:
1. `warningActions` -> `warningAction` (it's not a list)
2. Proposal to simply this if-else-if-... chain:

```kotlin
val settings = CodyApplicationSettings.instance
val warningAction = when {
    settings.autocompleteRateLimitError && settings.chatRateLimitError -> ChatAndAutocompleteLimitWarningAction()
    settings.autocompleteRateLimitError -> AutocompletionLimitWarningAction()
    settings.chatRateLimitError -> ChatLimitWarningAction()
    else -> null
}
```